### ISSUE TITLE
Incorporate user/item metadata

### ISSUE THREAD

**Issue by novan-p-simanjuntak** [maciejkula/spotlight#89]
Hi,
Is there a way to incorporate user metadata (e.g, age, location) and item (e.g, tag) metadata in spotlight?

**Comment by maciejkula** [comment]
Yes, if you write your own representation layers. There is nothing baked-in.

**Comment by rragundez** [comment]
this shouldn't be too difficult to implement, I do think it is quite useful specially for applications with limited data at the beginning where user / item metadata can serve to reduce the cold start problem of a recommendation system. We will probably need an extra layer as @maciejkula suggest, that deals with the preparation before sending it to the `Bilinear` layer, as the `Bilinear` later is very clear and simple on what it does and I think it should remain like that in it's `forward` method.

I could take a look if you think this is a nice feature to add @maciejkula
